native/components/rules.react.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+
+import type { StyleSheetOf } from '../themes/colors';
+import type { MarkdownStyles } from '../themes/markdown-styles';
+
+import * as React from 'react';
+import { Text, Linking } from 'react-native';
+import * as SimpleMarkdown from 'simple-markdown';
+
+import { onlyEmojiRegex } from 'lib/shared/emojis';
+
+export default function rules(styles: StyleSheetOf<MarkdownStyles>) {
+  return {
+    // simple-markdown uses autolink rule to catch links inside <>
+    // autolink node is a node of link type, it will match links inside <>
+    // inside of it's content will be text node, with target to be opened onPress
+    // we don't need to specify how it should be transformed into output, link will handle this
+    autolink: {
+      ...SimpleMarkdown.defaultRules.autolink,
+    },
+    // link node is responsible for [title](actual link) links
+    // it is also type for autolink and url nodes, that are going through it
+    // content of this node is a text node, styled based on state.linkPresent
+    // link contains additional field target, opened on press
+    link: {
+      ...SimpleMarkdown.defaultRules.link,
+      react(
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) {
+        // setting state.linkPresent to true to mark that link is found
+        // then we need to make it false, so we know that it's ended
+        // in one message we might have normal text after link
+        // so if we don't do this, in text rule we will not apply proper text style
+        // we're doing this only here, since autolink and url are going through link
+        state.linkPresent = true;
+        const onPressLink = () => {
+          Linking.openURL(node.target);
+        };
+        const link = (
+          <Text key={state.key} onPress={onPressLink}>
+            {output(node.content, state)}
+          </Text>
+        );
+        state.linkPresent = false;
+        return link;
+      },
+    },
+    // this is the main node, containing other nodes inside it's content
+    // if content is contains something more than just plain text (for example text, then some link)
+    // then it will be splited into separate nodes, with appropriate types (in this case, text and link)
+    // paragraph style is responsible for wrapping message bubble over text
+    paragraph: {
+      ...SimpleMarkdown.defaultRules.paragraph,
+      react(
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) {
+        return (
+          <Text key={state.key} style={styles.paragraph}>
+            {output(node.content, state)}
+          </Text>
+        );
+      },
+    },
```

Comment:
Eslint is complaining about missing display name when I'm trying to do this

Reply:
Yeah, ESLint thinks this function is a React component because it's a function that returns a `React.Node`. But in our case we never use it as a `<Component>`, so it's fine. Add an `eslint-ignore` line